index.js

```diff
@@ -23,11 +23,16 @@ class Fetcher {
 		this.headers = mergedOptions.headers;
 	}
 
-	async fetch({ method = 'GET', path, query = null, headers = {}, body = null }) {
+	async fetch({ method = 'GET', path, url, query = null, headers = {}, body = null }) {
+
+		if (!path && !url) {
+			throw new Error('No url or path provided, one is required');
+		}
+
 		headers = deepmerge(this.headers, headers);
 
-		const url = formatUrl({
-			pathname: `${this.baseUrl}${path}`,
+		const formattedUrl = formatUrl({
+			pathname: url ? url : `${this.baseUrl}${path}`,
```

Comment:
this looks weird to me. `pathname` is only supposed to be the path part. e.g. `/foo/bar`, you could use `url.parse` on the base url to get its components and then merge them somehow afterwards before calling `url.format` on the while thing again